src/components/Menu/MenuItem.tsx

```diff
@@ -110,6 +139,11 @@ class MenuItem extends UIComponent<Extendable<MenuItemProps>, MenuItemState> {
     underlined: PropTypes.bool,
     vertical: PropTypes.bool,
     wrapper: PropTypes.oneOfType([PropTypes.node, PropTypes.object]),
+    menu: customPropTypes.itemShorthand,
+    submenuOpen: PropTypes.bool,
+    defaultSubmenuOpen: PropTypes.bool,
+    setActiveIndex: PropTypes.func,
```

Comment:
`index` is only used for making this item `active`. We can remove `index` altogether and rename this prop to something that is more specific (and better reflects intent of this method), like `makeSelfActive` or `activate`